test/integration/etcd/data.go

```diff
@@ -64,344 +66,417 @@ func GetEtcdStorageDataForNamespace(namespace string) map[schema.GroupVersionRes
 // It is exported so that it can be reused across multiple tests.
 // It returns a new map on every invocation to prevent different tests from mutating shared state.
 // Namespaced objects keys are computed for the specified namespace.
-func GetEtcdStorageDataForNamespaceServedAt(namespace string, version string, removeAlphas bool) map[schema.GroupVersionResource]StorageData {
+func GetEtcdStorageDataForNamespaceServedAt(namespace string, v string, removeAlphas bool) map[schema.GroupVersionResource]StorageData {
 	image := image.GetE2EImage(image.BusyBox)
 	etcdStorageData := map[schema.GroupVersionResource]StorageData{
 		// k8s.io/kubernetes/pkg/api/v1
 		gvr("", "v1", "configmaps"): {
-			Stub:             `{"data": {"foo": "bar"}, "metadata": {"name": "cm1"}}`,
-			ExpectedEtcdPath: "/registry/configmaps/" + namespace + "/cm1",
+			Stub:              `{"data": {"foo": "bar"}, "metadata": {"name": "cm1"}}`,
+			ExpectedEtcdPath:  "/registry/configmaps/" + namespace + "/cm1",
+			IntroducedVersion: "1.2",
```

Comment:
to minimize the diff, would it make sense to only add this for APIs added / removed >=1.31 where emulated version is relevant, and assume APIs without these explicitly set are available in all versions?

Reply:
I included the information so that future API additions have a proper reference and we can properly lint for the fields being present. This technically extends to almost APIs added since 1.25 since our 6 cycle removal policy would point to 1.31 or later removal. I don't think the diff is that much larger when reviewing with "hide whitespace changes" and the versions were programmatically generated. Open to removing if you still feel like it's a big change though.

Reply:
it'll change the blame on ~all the lines in the data file, which is a touch annoying, and I didn't really want to review / verify all the version lines that don't really matter, but I guess it's not a huge deal